### ISSUE TITLE
star (*) symbol in text box cuts off bottom of text when saved

### ISSUE THREAD

**Issue by christopherlovell** [matplotlib/matplotlib#15313]
### Bug report

**Bug summary**

When using the star (*) symbol in a `text` box, it cuts off the bottom of the text when saved (`savefig`). This is only the case when `usetex=False` in `rcParams`.

**Code for reproduction**

```python
import matplotlib
import matplotlib.pyplot as plt

from matplotlib import rc
rc('text', usetex=False)

fig, ax = plt.subplots(1,1)
ax.text(0.5, 0.5, '$S_{*} M^{ath}$ and some not math',
         transform=ax.transAxes)
plt.savefig('test1.png', bbox_inches='tight')
```

![test2](https://user-images.githubusercontent.com/4648092/65336879-07f14680-dbbf-11e9-9a0f-43fc495b20cc.png)

You can see the bottom of the text box is cut off. Whereas this plots fine:

```python
fig, ax = plt.subplots(1,1)
ax.text(0.5, 0.5, '$S_{ome} M^{ath}$ and some not math',
         transform=ax.transAxes)
plt.savefig('test1.png', bbox_inches='tight')
```

![test1](https://user-images.githubusercontent.com/4648092/65336976-32db9a80-dbbf-11e9-98bf-306e5166ad9b.png)

**Matplotlib version**
  * Operating system: Unix
  * Matplotlib version: 3.1.1
  * Matplotlib backend (`print(matplotlib.get_backend())`): TkAgg
  * Python version: Python 3.6.8 (default, Apr 25 2019, 21:02:35)

matplotlib installed from pip

  * Other libraries: 

cycler==0.10.0
kiwisolver==1.1.0
numpy==1.17.2
pyparsing==2.4.2
python-dateutil==2.8.0
six==1.12.0

**Comment by ImportanceOfBeingErnest** [comment]
The first figure is much larger than the second. How exactly are they produced?

**Comment by jklymak** [comment]
I can confirm this on my machine. 

```python 
import matplotlib.pyplot as plt
fig, ax = plt.subplots(1,1)
ax.text(0.5, 0.5, '$S_{x} Math$',
         transform=ax.transAxes)
ax.text(0.5, 0.2, '$S_{*} Math$',
         transform=ax.transAxes)
ax.text(0.2, 0.35, '$S_{*} Math_{p}$',
         transform=ax.transAxes)
plt.savefig('test1.png')
```
![test1](https://user-images.githubusercontent.com/1562854/65376208-1d638080-dc52-11e9-9275-0520f2065b17.png)

It looks like somehow the lower extent is being set by the subscript if there is one, and not the minimum of the subscript and the rest of the text.  Note if you put a bigger subscript in, all is well.

**Comment by ImportanceOfBeingErnest** [comment]
So what's the difference? 
The font seems to be the same. 
I'm using windows instead of linux. 
Maybe version of freetype? (I have freetype 2.9.1 ha9979f8_1 here)

**Comment by ImportanceOfBeingErnest** [comment]
Oh, it's the dpi. Would be good to state it explicitely in the code, because matplotlib's default dpi=100 works fine it seems.

**Comment by jklymak** [comment]
I dunno.  So far, `$S Math$` has a height of 26, and descent of 2.  `$S_{*} Math$` has a height of 21.2 and descent of 2.  So somehow the height is being miscalculated.

**Comment by jklymak** [comment]
Not sure why it ends up OK for 100 dpi, but yes thats true.  Is matplotlib's default really 100 dpi? Regardless, the font metrics are incorrect so thats probably the root of the problem

**Comment by jklymak** [comment]
Interestingly, this gets worse and worse the higher the dpi: 

1500 dpi:

<img width="551" alt="offset" src="https://user-images.githubusercontent.com/1562854/65380950-5f64e480-dc9b-11e9-825e-495fa8b3f55a.png">

This is not a problem for pdf backend, so its just agg/png.

**Comment by ImportanceOfBeingErnest** [comment]
It seems that at this point

https://github.com/matplotlib/matplotlib/blob/27931d4cf3bcc5d2f5f9a7135b9edfa955d70a20/lib/matplotlib/mathtext.py#L200

`self.height` is wrong. But I'm unsure why.

**Comment by anntzer** [comment]
fwiw this doesn't happen with mplcairo either; I have some plan to just lift the mplcairo mathtext code to replace MathtextBackendAgg...

**Comment by ImportanceOfBeingErnest** [comment]
Isn't it that Cairo can directly render the glyphs, while Agg needs a FTImage? How would you replace that? (Matplotlib's internal Cairo backends also work fine in this context here.)

**Comment by anntzer** [comment]
It doesn't really matter whether the rendering is done directly or via FT2Image; I *think* the important point is to repro the logic of rounding fractional glyph positions.

**Comment by ImportanceOfBeingErnest** [comment]
The issue here is unrelated to rounding. It seems to be in the logic that places the image. If you replace this line

https://github.com/matplotlib/matplotlib/blob/27931d4cf3bcc5d2f5f9a7135b9edfa955d70a20/lib/matplotlib/mathtext.py#L194

by 

    (bbox[3] - bbox[1]),

i.e. don't subtract the "original depth", the result will be


![image](https://user-images.githubusercontent.com/23121882/65396915-cbcf0900-ddab-11e9-8ba2-45c061084a84.png)

which shows the correct image for the text that does not go below the baseline. But somehow you need to account for the depth, or else the image where text does go below the baseline will be placed wrongly aligned.

**Comment by jklymak** [comment]
So should it just be that oy is changed?

**Comment by ImportanceOfBeingErnest** [comment]
I don't think so, because I wouldn't know which quantity to use to discriminate the two cases. But maybe someone else has an idea, in which case <details>
 <summary>this is the code to reproduce above images.</summary>

```
import matplotlib.pyplot as plt
plt.rcParams["font.size"] = 48
fig, ax = plt.subplots(1,1, figsize=(5.5,2.7), dpi=300)
fig.subplots_adjust(left=0, right=1)
ax.axis("off")
t4 = ax.text(0.05, 0.5, '$S l$', transform=ax.transAxes)
t3 = ax.text(0.25, 0.5, '$S_{x} l$', transform=ax.transAxes)
t1 = ax.text(0.5, 0.5, '$S_{°} l$', transform=ax.transAxes)
t2 = ax.text(0.75, 0.5, '$S_{*} l$', transform=ax.transAxes)

ax.axhline(0.5, color="gray", linewidth=0.4)
plt.savefig('test1.png')
plt.savefig('test1.pdf')
plt.show()
```

</details>

**Comment by jklymak** [comment]
Yeah somehow the baseline has been set to the bottom of the subscript but I’ve not been able to track how that gets decided. Maybe in the image creation?

**Comment by jklymak** [comment]
An unhelpful data point: this has been like this since at least 2.0.  I have trouble compiling anything older....

**Comment by ImportanceOfBeingErnest** [comment]
It was also present earlier, but in 1.x the default math font was computer modern, which puts the asterix below the baseline and hence one wouldn't have noticed.

**Comment by anntzer** [comment]
See #15339 (still needs some work and description).

**Comment by anntzer** [comment]
Fixed by the switch to direct rendering without an intermediate buffer.

**Comment by bshkknd** [comment]
To resolve a similar issue, I found it helpful to use the “\ast” character instead of the “*” character. This change ensured proper text rendering without any clipping problems.
